pinot-broker/src/main/java/com/linkedin/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -78,13 +93,16 @@
 
   public BaseBrokerRequestHandler(Configuration config, RoutingTable routingTable,
       TimeBoundaryService timeBoundaryService, AccessControlFactory accessControlFactory,
-      TableQueryQuotaManager tableQueryQuotaManager, BrokerMetrics brokerMetrics) {
+      TableQueryQuotaManager tableQueryQuotaManager, ZkHelixPropertyStore<ZNRecord> propertyStore,
+      BrokerMetrics brokerMetrics) {
     _config = config;
     _routingTable = routingTable;
     _timeBoundaryService = timeBoundaryService;
     _accessControlFactory = accessControlFactory;
     _tableQueryQuotaManager = tableQueryQuotaManager;
+    _propertyStore = propertyStore;
     _brokerMetrics = brokerMetrics;
+    _tableSchemaCache = new TableSchemaCache(_propertyStore);
```

Comment:
It might be worth it to keep the tableSchemaCache as a top level citizen that gets passed in (instead of property-store). Its possible classes outside of the RequestHandler might need access to it.

Reply:
Done